datasets.py

```diff
@@ -1,9 +1,16 @@
 from typing import *
+from pathlib import Path
```

Comment:
Can we just use `os.path` so we won't need an extra library?

Reply:
Don't sleep on `Path`s. I much prefer using `pathlib` and `shutil` to `os`, they're both built-in libs just like `os` anyway